src/jsonValidation/appIntroValidator.ts

```diff
@@ -0,0 +1,34 @@
+import { isArray, isObjectLike, isString } from 'lodash';
+
+import { Initializer, Initializers } from '../helpers/initializationHelper';
+import { AppIntroSlide } from '../types';
+
+type AppIntroSlideData = {
+  image: string;
+  title: string;
+  text: string;
+  onLeaveSlide?: Initializer;
+};
+
+const isValidAppIntroSlideData = (json: unknown): json is AppIntroSlideData => {
+  if (!isObjectLike(json)) {
+    return false;
+  }
+
+  const { image, title, text, onLeaveSlide } = json as AppIntroSlideData;
+
+  const isInitializer = !onLeaveSlide || Object.values(Initializer).includes(onLeaveSlide);
```

Comment:
could this line get a comment please.
trying to get the context from code is not that easy.
<img width="431" alt="Bildschirmfoto 2021-11-24 um 15 19 14" src="https://user-images.githubusercontent.com/1942953/143255306-679a939d-4fab-4c67-bbfb-d93cc068f284.png">
<img width="530" alt="Bildschirmfoto 2021-11-24 um 15 19 21" src="https://user-images.githubusercontent.com/1942953/143255315-7eb76836-57d7-4bc9-89b4-9893ac0e06f3.png">

i guess the values of `Initializer` are:
- 'locationService'
- 'matomoTracking'
- 'pushNotifications

so we check here if a `onLeaveSlide` is on of them?

Reply:
yes. the purpose is to filter out slides that have invalid `onLeaveSlide`s in them. that way slides corresponding to features of newer app versions will not be shown.

Reply:
> could this line get a comment please.

do you mind adding your sentence as comment there?

Reply:
i do it

Reply:
did it